mlir/lib/Analysis/Presburger/PresburgerSet.cpp

```diff
@@ -395,39 +395,170 @@ Optional<uint64_t> PresburgerSet::computeVolume() const {
   return result;
 }
 
-PresburgerSet PresburgerSet::coalesce() const {
-  PresburgerSet newSet =
-      PresburgerSet::getEmptySet(getNumDimIds(), getNumSymbolIds());
-  llvm::SmallBitVector isRedundant(getNumPolys());
+bool containedFacet(ArrayRef<int64_t> ineq, IntegerPolyhedron p,
+                    ArrayRef<ArrayRef<int64_t>> cut) {
+  Simplex simp(p);
+  simp.addEquality(ineq);
```

Comment:
This inequality already holds, so I could achieve the same by adding its inverse as an inequality. Does this have any performance implications?